spec/lib/insert/insert_refunds_spec.rb

```diff
@@ -4,82 +4,192 @@
 # Full license explanation at https://github.com/houdiniproject/houdini/blob/master/LICENSE
 require 'rails_helper'
 
-describe InsertRefunds, pending: true do
-  # before(:all) do
-  #   stripe_acct = VCR.use_cassette('InsertRefunds/stripe_acct'){Stripe::Account.create(managed: true, country: 'US', email: 'uzr@example.com').id}
-  #  @supp = Psql.execute(Qexpr.new.insert('supporters', [{name: 'nnname'}])).first
-  #  @payment = Psql.execute(Qexpr.new.insert('payments', [{supporter_id: @supp['id'], gross_amount: 100, fee_total: 33, net_amount: 67, refund_total: 0}]).returning('*')).first
-  #  @nonprofit = Psql.execute(Qexpr.new.insert('nonprofits', [{name: 'xxxx', stripe_account_id: stripe_acct}]).returning('*')).first
-  #   stripe_token  = VCR.use_cassette('InsertRefunds/stripe_token'){Stripe::Token.create(card: StripeTestHelpers::Card_immediate).id}
-  #   stripe_charge = VCR.use_cassette('InsertRefunds/stripe_charge'){Stripe::Charge.create(amount: 10000, currency: 'usd', source: stripe_token, description: 'charge 1', destination: stripe_acct)}
-  #  @charge = Qx.insert_into(:charges).values({
-  #    amount: 100,
-  #    stripe_charge_id: stripe_charge.id,
-  #    nonprofit_id: @nonprofit['id'],
-  #    payment_id: @payment['id'],
-  #    supporter_id: @supp['id']
-  #  }).returning('*').timestamps.execute.first
-  #  @refund_amount = 100
-  #  @fees = CalculateFees.for_single_amount(100)
-  # end
+describe InsertRefunds do
```

Comment:
@wwahammy, when I started to work on the tests for the event publishing on `InsertRefunds.with_stripe`, I saw that those tests were pending for a long time, so I tried to update them a little to work with the current behavior. I think the calculations for the refund aren't right so I just "xit" the tests, but I can update the class to work properly since I'm already working on things here.

Reply:
@clarissalimab This is a good idea to get these tests working! 

I'm not sure I understand exactly what you're saying. In particular:

* What calculations aren't right?
* When you say "I can update the class to work properly since I'm already working on things here" what does this all refer to?

Reply:
> What calculations aren't right?

I think the calculation for the refund isn't updated, actually I'm not sure if it isn't right, but I looked at the calculation in CC and it's different.

> When you say "I can update the class to work properly since I'm already working on things here" what does this all refer to?

I had to add some code to it to make it work with transactions. I meant that if the calculation is wrong I can work on it to have it working as we want it to.

Reply:
Hmmm. I'm still not sure I understand what 'calculation for the refund' you're referring to.

Reply:
Because when we do the refund, there is a calculation for the fee, I thought it could be outdated because the tests were pending (and failing) for a long time. I updated the tests to work with the current behavior, but I think the correct thing to do would be to understand how the calculations should be and write the tests with that in mind instead.